test/openssl/test_ts.rb

```diff
@@ -181,6 +181,12 @@ def test_request_encode_decode
     assert_equal(42, qer2.nonce)
   end
 
+  def test_request_invalid_asn1
+    assert_raise_with_message(OpenSSL::Timestamp::TimestampError, /nested asn1 error/) do
```

Comment:
Can you check the exception type only (`assert_raise(OpenSSL::Timestamp::TimestampError)`)?

"nested asn1 error" message sounds like OpenSSL's implementation detail and I feel it's likely to change in future releases.